driver/registry.go

```diff
@@ -107,15 +109,23 @@ type selfServiceStrategy interface {
 }
 
 func NewRegistry(c configuration.Provider) (Registry, error) {
-	driver, err := dbal.GetDriverFor(c.DSN())
+	driver, err := dbal.GetDriverFor(c.DSNAddress())
 	if err != nil {
-		return nil, err
+		return nil, errors.WithStack(err)
 	}
 
 	registry, ok := driver.(Registry)
 	if !ok {
 		return nil, errors.Errorf("driver of type %T does not implement interface Registry", driver)
 	}
 
+	// if dsn is memory we have to run the migrations on every start
+	if c.DSN() == "memory" {
```

Comment:
Let's do this for all DSN that have prefix `sqlite://` and a URL query with `mode=memory`, that way this also works with custom SQLite DSN strings that do not use `memory` (could be useful for testing other SQLite modifiers for example).

Also removes the need for another function `DSNAddress`